frontend/javascripts/oxalis/throttled_store.js

```diff
@@ -17,14 +17,14 @@ Store.subscribe(() => {
   if (state !== prevState) {
     prevState = state;
     waitForUpdate.resolve();
-    waitForUpdate = new Deferred();
   }
 });
 
 async function go() {
   // eslint-disable-next-line no-constant-condition
   while (true) {
     await waitForUpdate.promise();
+    waitForUpdate = new Deferred();
```

Comment:
This fixes a bug where store updates could be swallowed if they were triggered by a store listener, themselves. In this case, this `go` loop would still be in line 30 executing the listeners (triggering a store change) which would cause the promise in line 19 to be resolved and then immediately reassigned. After executing the listeners, the code would then wait for a store change in line 27, although the change already happened.
I tested that my change fixes this scenario.